File: examples/php/sort_1.php
<?php
function bubble_sort($arr) {
    $n = count($arr);
    for ($i = 0; $i < $n; $i++) {
        for ($j = 0; $j < $n - 1 - $i; $j++) {
            if ($arr[$j] > $arr[$j + 1]) {
                list($arr[$j], $arr[$j + 1]) = [$arr[$j + 1], $arr[$j]];
            }
        }
    }
    return $arr;
}
$data = [64, 34, 25, 12, 22];
echo implode(',', bubble_sort($data)) . "\n";

File: examples/ruby/sort_1.rb
def bubble_sort(arr)
  n = arr.size
  (0...n).each do |i|
    (0...n - 1 - i).each do |j|
      arr[j], arr[j + 1] = arr[j + 1], arr[j] if arr[j] > arr[j + 1]
    end
  end
  arr
end

data = [64, 34, 25, 12, 22]
puts bubble_sort(data.dup).inspect
